src/ensembl/src/content/app/custom-download/containers/content/attributes-accordion/state/attributesAccordionActions.ts

```diff
@@ -0,0 +1,225 @@
+)<{ searchTerm: string }, {}, Error>();
+
+export const fetchOrthologueSpecies: ActionCreator<
+  ThunkAction<void, any, null, Action<string>>
+> = (
+  searchTerm: string,
+  shouldShowBestMatches: boolean,
+  shouldShowAll: boolean,
+  orthologueSpecies: any
+) => async (dispatch) => {
+  dispatch(setOrthologueSpecies.request({ searchTerm: searchTerm }));
+  try {
+    // This will be fetched from the API when we have one
+    let allSpecies = sampleOrthologueSpecies.species;
+
+    let filteredSpecies: any = {};
+
+    allSpecies.forEach((species: any) => {
+      if (
+        species.display_name.toLowerCase().indexOf(searchTerm.toLowerCase()) !==
+        -1
+      ) {
+        filteredSpecies[species.name] = {
+          id: species.name,
+          label: species.display_name,
+          checkedStatus: false
+        };
+      }
+    });
+
+    if (orthologueSpecies && orthologueSpecies.default) {
+      Object.keys(orthologueSpecies.default).forEach((species: string) => {
+        if (orthologueSpecies.default[species].checkedStatus) {
+          filteredSpecies[species] = orthologueSpecies.default[species];
+        }
+      });
+    }
+
+    dispatch(setOrthologueSpecies.success({ default: filteredSpecies }));
+  } catch (error) {
+    dispatch(setOrthologueSpecies.failure(error));
+  }
+};
+
+export const setLocationAttributes = createAction(
+  'custom-download/set-location-attributes',
+  (resolve) => {
+    return (locationAttributes: {}) =>
+      resolve(
+        locationAttributes,
+        getCustomDownloadAnalyticsObject('Location attributes updated')
+      );
+  }
+);
+
+export const setSomaticVariationAttributes = createAction(
+  'custom-download/set-somatic-variation-attributes',
+  (resolve) => {
+    return (variationAttributes: {}) =>
+      resolve(
+        variationAttributes,
+        getCustomDownloadAnalyticsObject('Somatic variation attributes updated')
+      );
+  }
+);
+
+export const setGermlineVariationAttributes = createAction(
+  'custom-download/set-germline-variation-attributes',
+  (resolve) => {
+    return (variationAttributes: {}) =>
+      resolve(
+        variationAttributes,
+        getCustomDownloadAnalyticsObject(
+          'Germline variation attributes updated'
+        )
+      );
+  }
+);
+
+export const setAttributesAccordionExpandedPanel = createAction(
+  'custom-download/set-attributes-accordion-expanded-panels',
+  (resolve) => {
+    return (expandedPanel: string) =>
+      resolve(
+        expandedPanel,
+        getCustomDownloadAnalyticsObject('Toggle attributes accordion panel')
+      );
+  }
+);
+
+export const setVariationAccordionExpandedPanels = createAction(
+  'custom-download/set-variation-attributes-accordion-expanded-panels',
+  (resolve) => {
+    return (expandedPanels: []) =>
+      resolve(
+        expandedPanels,
+        getCustomDownloadAnalyticsObject(
+          'Toggle variation attributes accordion panel'
+        )
+      );
+  }
+);
```

Comment:
Regarding all the analytics events in this file: they are very granular and related to interactions with different elements of the interface. Are all of these events needed? Is anyone going to analyze them?

Reply:
At the moment I am tracking almost all the actions that are dispatched within custom downloads. This is because I thought we might need something to track for the june-alpha release and we do not have enough time to wait for the spec and then add the tracking afterwards.